api/src/clojure/org/akvo/flow_api/boundary/user.clj

```diff
@@ -18,3 +19,8 @@
       (let [id (user/id email)]
         (put-id user-cache instance-id email id)
         id))))
+
+(defn id-by-email-or-throw-error [remote-api instance-id email]
```

Comment:
maybe use function arg overloading 🤔 
https://github.com/akvo/akvo-flow-api/pull/156/commits/2c7ab9eb38420ca83a2b83e09d2480381b378818

Reply:
Or just handle the logic in the call site and skip the somewhat oddly named fn "-or-throw-error".

Reply:
There are 4 call sites that require the same logic, so it seems ok-ish to leave it some common place.

The fn has a weird name because the fn has a weird semantics of doing some side effect to break the flow of control by throwing an exception. The name was chosen specifically to encourage to not use it.

I personally prefer to have this additional fn with this particular behavior instead of the proposed overloading, so that the special and non-functional code is separated from the actual `id-by-email`, but if anybody has a strong feeling that the overload version is better, I am ok with that version also.

Reply:
Ah, I only saw one call site when I searched :-/ . This was not a big issue for me. So if it was chosen with consideration go for it.

Reply:
I was pretending to have a similar style than https://github.com/dakrone/clj-http/blob/3.x/src/clj_http/client.clj#L248  `:throw-exceptions` option
... so trying to be more clojure friendly 
final choice is up to you @dlebrero  !